### PR TITLE
Issue #9563: update example of AST for TokenTypes.RESOURCE

### LINKED ISSUE #9563: update example of AST for TokenTypes.RESOURCE
https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#RESOURCE

needs to be extended to show example of AST.
Source to be updated: https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java

PLEASE READ for more details: https://github.com/checkstyle/checkstyle/wiki/Good-practice-on-how-to-update-AST-in-TokenTypes

### PR DESCRIPTION
**Pull request by MahmoudAbdelazim**
Closes #9563
image
image

Code used for test: 
```
mahmoud:~/Desktop/checkstyle$ more Test.java 
public class A {
    public void b {
    	try (Foo foo = new Foo(); Bar bar = new Bar()) { }
    } 
}
```
Generated AST:
```
mahmoud:~/Desktop/checkstyle$ java -jar checkstyle-8.41-all.jar -T Test.java 
CLASS_DEF -> CLASS_DEF [2:0]
|--MODIFIERS -> MODIFIERS [2:0]
|   `--LITERAL_PUBLIC -> public [2:0]
|--LITERAL_CLASS -> class [2:7]
|--IDENT -> A [2:13]
`--OBJBLOCK -> OBJBLOCK [2:15]
    |--LCURLY -> { [2:15]
    |--VARIABLE_DEF -> VARIABLE_DEF [3:4]
    |   |--MODIFIERS -> MODIFIERS [3:4]
    |   |   `--LITERAL_PUBLIC -> public [3:4]
    |   |--TYPE -> TYPE [3:11]
    |   |   `--LITERAL_VOID -> void [3:11]
    |   `--IDENT -> b [3:16]
    |--INSTANCE_INIT -> INSTANCE_INIT [3:18]
    |   `--SLIST -> { [3:18]
    |       |--LITERAL_TRY -> try [4:5]
    |       |   |--RESOURCE_SPECIFICATION -> RESOURCE_SPECIFICATION [4:9]
    |       |   |   |--LPAREN -> ( [4:9]
    |       |   |   |--RESOURCES -> RESOURCES [4:10]
    |       |   |   |   |--RESOURCE -> RESOURCE [4:10]
    |       |   |   |   |   |--MODIFIERS -> MODIFIERS [4:10]
    |       |   |   |   |   |--TYPE -> TYPE [4:10]
    |       |   |   |   |   |   `--IDENT -> Foo [4:10]
    |       |   |   |   |   |--IDENT -> foo [4:14]
    |       |   |   |   |   `--ASSIGN -> = [4:18]
    |       |   |   |   |       `--EXPR -> EXPR [4:20]
    |       |   |   |   |           `--LITERAL_NEW -> new [4:20]
    |       |   |   |   |               |--IDENT -> Foo [4:24]
    |       |   |   |   |               |--LPAREN -> ( [4:27]
    |       |   |   |   |               |--ELIST -> ELIST [4:28]
    |       |   |   |   |               `--RPAREN -> ) [4:28]
    |       |   |   |   |--SEMI -> ; [4:29]
    |       |   |   |   `--RESOURCE -> RESOURCE [4:31]
    |       |   |   |       |--MODIFIERS -> MODIFIERS [4:31]
    |       |   |   |       |--TYPE -> TYPE [4:31]
    |       |   |   |       |   `--IDENT -> Bar [4:31]
    |       |   |   |       |--IDENT -> bar [4:35]
    |       |   |   |       `--ASSIGN -> = [4:39]
    |       |   |   |           `--EXPR -> EXPR [4:41]
    |       |   |   |               `--LITERAL_NEW -> new [4:41]
    |       |   |   |                   |--IDENT -> Bar [4:45]
    |       |   |   |                   |--LPAREN -> ( [4:48]
    |       |   |   |                   |--ELIST -> ELIST [4:49]
    |       |   |   |                   `--RPAREN -> ) [4:49]
    |       |   |   `--RPAREN -> ) [4:50]
    |       |   `--SLIST -> { [4:52]
    |       |       `--RCURLY -> } [4:54]
    |       `--RCURLY -> } [5:4]
    `--RCURLY -> } [6:0]
```

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 1]
```diff
@@ -2335,6 +2335,57 @@
      * A resource in the Java 7 try-with-resources construct.
      * This is a child of RESOURCES.
      *
+     * <p>For example:</p>
+     * <pre>
+     * try (Foo foo = new Foo(); Bar bar = new Bar()) { }
+     * </pre>
+     * <p>parses as:</p>
+     * <pre>
+     * +--LITERAL_TRY (try)
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:2344]: Please use CLI output as is

**Review by romani — CHANGES_REQUESTED** [review]
Item:

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 1 above)
Comment (MahmoudAbdelazim) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:2344]: On it

Comment (MahmoudAbdelazim) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java]: Done, please check it again

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 2]
```diff
@@ -2380,6 +2380,46 @@
      * A resource in the Java 7 try-with-resources construct.
      * This is a child of RESOURCES.
      *
+     * <p>For example:</p>
+     * <pre>
+     * try (Foo foo = new Foo(); Bar bar = new Bar()) { }
+     * </pre>
+     * <p>parses as:</p>
+     * <pre>
+     * LITERAL_TRY -&gt; try
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:2389]: Please use `|--` to bind this token with `}`, all tokens should be connected

**Review by romani — CHANGES_REQUESTED** [review]
Item:

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 2 above)
Comment (MahmoudAbdelazim) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:2389]: On it.

Comment (MahmoudAbdelazim) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java]: Done

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 3]
```diff
@@ -2380,6 +2380,46 @@
      * A resource in the Java 7 try-with-resources construct.
      * This is a child of RESOURCES.
      *
+     * <p>For example:</p>
+     * <pre>
+     * try (Foo foo = new Foo(); Bar bar = new Bar()) { }
+     * </pre>
+     * <p>parses as:</p>
+     * <pre>
+     * |--LITERAL_TRY -&gt; try
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:2389]: Each line should lead to somewhere.
Please remove `|--`

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 4]
```diff
@@ -2380,6 +2380,46 @@
...
+     *     |   |   |       `--ASSIGN -&gt; =
+     *     |   |   |           `--EXPR -&gt; EXPR
+     *     |   |   |               `--LITERAL_NEW -&gt; new
+     *     |   |   |                   |--IDENT -&gt; Bar
+     *     |   |   |                   |--LPAREN -&gt; (
+     *     |   |   |                   |--ELIST -&gt; ELIST
+     *     |   |   |                   `--RPAREN -&gt; )
+     *     |   |   `--RPAREN -&gt; )
+     *     |   `--SLIST -&gt; {
+     *     |       `--RCURLY -&gt; }
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:2420]: Remove whole vertical `|`

**Review by romani — COMMENTED** [review]
Items:

**Review by romani — CHANGES_REQUESTED** [review]
See above

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 4 above)
Comment (MahmoudAbdelazim) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:2420]: Ok, I just misunderstood the previous requested change, I'll fix it now.

Comment (MahmoudAbdelazim) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java]: I think it's now identical to the CLI output, please check it again

**Review by romani — APPROVED** [review]

**Pull request merged**